src/lxc/confile.c

```diff
@@ -1562,6 +1564,61 @@ static int set_config_sysctl(const char *key, const char *value,
 	return -1;
 }
 
+static int set_config_proc(const char *key, const char *value,
+			    struct lxc_conf *lxc_conf, void *data)
+{
+	char *subkey;
+	char *token = "lxc.proc.";
+	struct lxc_list *proclist = NULL;
+	struct lxc_proc *procelem = NULL;
+
+	if (lxc_config_value_empty(value))
+		return clr_config_proc(key, lxc_conf, NULL);
+
+	subkey = strstr(key, token);
+	if (!subkey)
+		return -1;
```

Comment:
Ah, I see you took the `set_config_cgroup()` call as inspiration. :) I don't like the `strstr()` that is used in there. This can all be done without using `strlen()` and friends. I mean, any reasonable compiler should optimize this anyway but at the same time I like to see it done in the code correctly without relying on the compiler being smart. :) So this should likely be:

```
---
 src/lxc/confile.c | 23 +++++++++--------------
 1 file changed, 9 insertions(+), 14 deletions(-)

diff --git a/src/lxc/confile.c b/src/lxc/confile.c
index 78c09f32..9df885b8 100644
--- a/src/lxc/confile.c
+++ b/src/lxc/confile.c
@@ -1567,40 +1567,35 @@ on_error:
 static int set_config_proc(const char *key, const char *value,
 			    struct lxc_conf *lxc_conf, void *data)
 {
-	char *subkey;
-	char *token = "lxc.proc.";
+	const char *subkey;
 	struct lxc_list *proclist = NULL;
 	struct lxc_proc *procelem = NULL;
 
 	if (lxc_config_value_empty(value))
 		return clr_config_proc(key, lxc_conf, NULL);
 
-	subkey = strstr(key, token);
-	if (!subkey)
+	if (strncmp(key, "lxc.proc.", sizeof("lxc.proc.") -1) != 0)
 		return -1;
 
-	if (!strlen(subkey))
-		return -1;
+	subkey = key + sizeof("lxc.proc.") - 1;
 
-	if (strlen(subkey) == strlen(token))
-		return -1;
-
-	subkey += strlen(token);
+	if (*subkey != '\0')
+		return -EINVAL;
 
 	proclist = malloc(sizeof(*proclist));
 	if (!proclist)
-		goto out;
+		goto on_error;
 
 	procelem = malloc(sizeof(*procelem));
 	if (!procelem)
-		goto out;
+		goto on_error;
 	memset(procelem, 0, sizeof(*procelem));
 
 	procelem->filename = strdup(subkey);
 	procelem->value = strdup(value);
 
 	if (!procelem->filename || !procelem->value)
-		goto out;
+		goto on_error;
 
 	proclist->elem = procelem;
 
@@ -1608,7 +1603,7 @@ static int set_config_proc(const char *key, const char *value,
 
 	return 0;
 
-out:
+on_error:
 	free(proclist);
 	if (procelem) {
 		free(procelem->filename);
-- 
2.14.1
```

Reply:
```
-	subkey += strlen(token);
+	if (*subkey != '\0')
+		return -EINVAL;
```
Is this should be 
```
-	subkey += strlen(token);
+	if (*subkey == '\0')
+		return -EINVAL;
```